### ISSUE TITLE
Deprecation warning for keys.makePublicKeyString(rsaKey)  is misleading

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4265]
|<img alt="acidtonic's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| acidtonic reported|
|-|-|
|Trac ID|trac#4265|
|Type|defect|
|Created|2010-02-09 18:03:24Z|
|Branch|https://github.com/twisted/twisted/tree/clearer-keys-deprecations-4265|

Deprecation warning says to use Key.toString() even for the public key deprecation warning....

Should be Key.public().toString() for public keys. It shows the same error for both which tripped me up.

Attachments:

* [makepublickeystring-deprecation.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/232/232aa8fdec7dba8e16449e85cfc5077b48349e19/8e3d28ef1ef1a65fef48c8a5f01605a50cc86990.patch) (747 bytes) - added by exarkun on 2010-02-09 18:38:52Z - Fix the suggestion in the deprecation warning
* [trial-test_keys-output.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/232/232aa8fdec7dba8e16449e85cfc5077b48349e19/05cf47802dd24f5ea9ba72bd8777135f75bf0052.txt) (12647 bytes) - added by jesstess on 2010-02-11 22:32:07Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4265 4265
type__defect defect
reporter__acidtonic acidtonic
priority__normal normal
milestone__ 
branch__branches_clearer_keys_deprecations_4265 branches/clearer-keys-deprecations-4265
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1265738604000000 1265738604000000
changetime__1265986242000000 1265986242000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_2"></a>|@jesstess set owner to @exarkun|
|-|-|

test_makePublicKeyStringErrors needs to be updated with the new deprecation warning.

Tangentially, I had never run these tests before on my machines and installed pycrypto and pyasn1 for this review to discover that several of the tests FAIL or ERROR. Do any of the build machines actually run these tests?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [28156]) Branching to 'clearer-keys-deprecations-4265'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

Woops.  I must have been switched to the wrong branch when I ran the tests.

Fixed and committed to a branch.  I also read some of the other deprecation warnings and found one that I think wasn't ideal, so I adjusted it as well.

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Oops.  As far as the question about build machines goes, many of them do have the necessary dependencies and do run the tests.  It might be worth investigating the failures you saw.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess commented|
|-|-|

I attached a transcript of the trial output on test_keys. The test was run on trunk (currently r28173). This is an Ubuntu machine using Python 2.6.2, PyCrypto 2.1.0, and pyasn1 revision 0.0.11a.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set owner to @exarkun|
|-|-|

I don't know what's up with the test failures I'm seeing (what versions of the crypto libraries are the build machines using?), but they're not related to this ticket.

Looks good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

Version information is available for each build.  If you go to a build overview page and then look at the 2nd step, you'll see a "versions" link.  If there's more information that you'd like to see there, let me know and I can add it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [28175]) Merge clearer-keys-deprecations-4265

Author: exarkun
Reviewer: jesstess
Fixes: [#4265](https://github.com/twisted/twisted/issues/4265)

Change two of the deprecation warnings for `twisted.conch.ssh.keys` functions
to be more accurate in their description of the appropriate replacement.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
